pkg/controller/clone/planner.go

```diff
@@ -431,6 +497,125 @@ func (p *Planner) planHostAssistedFromPVC(ctx context.Context, args *PlanArgs) (
 	return []Phase{hcp, rp}, nil
 }
 
+func (p *Planner) planHostAssistedFromSnapshot(ctx context.Context, args *PlanArgs) ([]Phase, error) {
+	sourceSnapshot := &snapshotv1.VolumeSnapshot{}
+	exists, err := getResource(ctx, p.Client, args.DataSource.Namespace, args.DataSource.Spec.Source.Name, sourceSnapshot)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("source claim does not exist")
+	}
+
+	sourceClaimForDumbClone, err := createTempSourceClaim(ctx, args.DataSource.Namespace, args.TargetClaim, sourceSnapshot, p.Client)
+	if err != nil {
+		return nil, err
+	}
+	cfsp := &SnapshotClonePhase{
+		Owner:          args.TargetClaim,
+		Namespace:      args.DataSource.Namespace,
+		SourceName:     args.DataSource.Spec.Source.Name,
+		DesiredClaim:   sourceClaimForDumbClone,
+		OwnershipLabel: p.OwnershipLabel,
+		Client:         p.Client,
+		Log:            args.Log,
+		Recorder:       p.Recorder,
+	}
+
+	pcp := &PrepClaimPhase{
+		Owner:           args.TargetClaim,
+		DesiredClaim:    sourceClaimForDumbClone.DeepCopy(),
+		Image:           p.Image,
+		PullPolicy:      p.PullPolicy,
+		InstallerLabels: p.InstallerLabels,
+		OwnershipLabel:  p.OwnershipLabel,
+		Client:          p.Client,
+		Log:             args.Log,
+		Recorder:        p.Recorder,
+	}
+
+	desiredClaim := createDesiredClaim(args.DataSource.Namespace, args.TargetClaim)
+	if util.ResolveVolumeMode(desiredClaim.Spec.VolumeMode) == corev1.PersistentVolumeFilesystem {
+		ds := desiredClaim.Spec.Resources.Requests[corev1.ResourceStorage]
+		is, err := cc.InflateSizeWithOverhead(ctx, p.Client, ds.Value(), &args.TargetClaim.Spec)
```

Comment:
@mhenriks I hope you are prepared to handle the inflation when integrating dvs with clone populators

Reply:
Can't wait